ReactViewControl.Avalonia/ReactViewRender.Avalonia.cs

```diff
@@ -12,20 +13,29 @@ partial class ReactViewRender : ContentControl, IStyleable {
 
         Type IStyleable.StyleKey => typeof(ContentControl);
 
-        partial void ExtraInitialize() {
+        private Window GetHiddenWindow() {
             if (hiddenWindow == null) {
                 hiddenWindow = new Window() {
                     IsVisible = false,
                     Focusable = false,
                     Title = "Hidden React View Window"
                 };
             }
+            return hiddenWindow;
+        }
 
-            WebView.HostingWindow = hiddenWindow;
-            WebView.AllowNativeMethodsParallelExecution = !ForceNativeSyncCalls;
+        partial void ExtraInitialize() {
             Content = WebView;
         }
 
+        partial void PreloadWebView() {
+            var window = GetHiddenWindow();
+            // initialize browser with full screen size to avoid html measure issues on initial render
+            var initialBrowserSizeWidth = (int)window.Screens.All.Max(s => s.WorkingArea.Width * (WebView.OsrEnabled ? 1 : s.PixelDensity));
```

Comment:
Shouldn't we use primary screen widht/height like WPF?

Reply:
Actually it should be the other way around but since there's no easy way to do that on wpf... I opted for the primary.